### PR TITLE
Fix extract_function_name

### PR DESCRIPTION
**Pull request by serena-ruan**
The fix is to make sure the function name strictly follows the syntax in the sql body and we correctly extract them.

### CODE REVIEW

**Comment by serena-ruan** [comment]
https://github.com/databricks-eng/ai-oss-integration-tests-runner/actions/runs/11287027455 passed

File: src/ucai/core/databricks.py [hunk 1]
```diff
@@ -89,22 +93,31 @@ def extract_function_name(sql_body: str) -> str:
...
     if match:
-        return match.group(1)
+        result = match.group(1)
+        full_function_name = validate_full_function_name(result)
+        # backticks are only required in SQL, not in python APIs
+        return (
+            f"{strip_backticks(full_function_name.catalog_name)}."
+            f"{strip_backticks(full_function_name.schema_name)}."
+            f"{strip_backticks(full_function_name.function_name)}"
+        )
```
Comment (harupy) [inline src/ucai/core/databricks.py]: I'd add a method to `FullFunctionName`. Imagine you need to do the same thing in other places.

File: src/ucai/core/utils/validation_utils.py [hunk 2]
```diff
@@ -29,6 +29,12 @@ def validate_full_function_name(function_name: str) -> FullFunctionName:
     return FullFunctionName(catalog_name=splits[0], schema_name=splits[1], function_name=splits[2])
 
 
+def strip_backticks(s: str) -> str:
+    if s.startswith("`") and s.endswith("`"):
+        return s[1:-1]
+    return s
```
Comment (harupy) [inline src/ucai/core/utils/validation_utils.py]: do we need this function? can we just use ``strip("`")``?

File: src/ucai/core/databricks.py (hunk 1 above)
Comment (harupy) [reply src/ucai/core/databricks.py]: can we strip `_name` in each field? That seems redundant.

**Review by harupy — APPROVED** [review]
LGTM

File: src/ucai/core/utils/function_processing_utils.py [hunk 3]
```diff
@@ -101,7 +101,7 @@ def get_tool_name(func_name: str) -> str:
     # OpenAI has constriant on the function name:
     # Must be a-z, A-Z, 0-9, or contain underscores and dashes, with a maximum length of 64.
     full_func_name = validate_full_function_name(func_name)
-    tool_name = f"{full_func_name.catalog_name}__{full_func_name.schema_name}__{full_func_name.function_name}"
+    tool_name = f"{full_func_name.catalog}__{full_func_name.schema}__{full_func_name.function}"
```
Comment (harupy) [inline src/ucai/core/utils/function_processing_utils.py]: can we make this as a method of `FullFunctionName`?

File: src/ucai/core/databricks.py [hunk 4]
```diff
@@ -89,22 +92,27 @@ def extract_function_name(sql_body: str) -> str:
...
     """
+    # NOTE: catalog/schema/function names follow guidance here:
+    # https://docs.databricks.com/en/sql/language-manual/sql-ref-names.html#catalog-name
     pattern = re.compile(
         r"""
         CREATE\s+(?:OR\s+REPLACE\s+)?      # Match 'CREATE OR REPLACE' or just 'CREATE'
         (?:TEMPORARY\s+)?                  # Match optional 'TEMPORARY'
         FUNCTION\s+(?:IF\s+NOT\s+EXISTS\s+)?  # Match 'FUNCTION' and optional 'IF NOT EXISTS'
-        ([\w.]+)                           # Capture the function name (including schema if present)
+        ([^ /.]+\.[^ /.]+\.[^ /.]+)          # Capture the function name (including schema if present)
```
Comment (harupy) [inline src/ucai/core/databricks.py]: In a follow-up, can we use named groups: https://docs.python.org/3.12/library/re.html?


`group(1)`: where is the first group?
`group("name")`: can easily locate where the name group is.

File: src/ucai/core/utils/validation_utils.py [hunk 5]
```diff
@@ -6,9 +6,12 @@
...
-    function_name: str
+    catalog: str
+    schema: str
+    function: str
+
+    def __str__(self) -> str:
+        return f"{self.catalog.strip('`')}.{self.schema.strip('`')}.{self.function.strip('`')}"
 
 
 def validate_full_function_name(function_name: str) -> FullFunctionName:
```
Comment (harupy) [inline src/ucai/core/utils/validation_utils.py]: can we make this a class method of `FullFunctionName`?

**Pull request merged**